src/wiktextract/extractor/de/example.py

```diff
@@ -0,0 +1,93 @@
+from collections import defaultdict
+from typing import Dict, List
+
+
+from wikitextprocessor import NodeKind, WikiNode
+from wikitextprocessor.parser import LevelNode
+from wiktextract.extractor.de.utils import find_and_remove_child, match_senseid
+
+from wiktextract.page import clean_node
+from wiktextract.wxr_context import WiktextractContext
+
+
+def extract_examples(
+    wxr: WiktextractContext,
+    page_data: List[Dict],
+    level_node: LevelNode,
+) -> None:
+    for list_node in level_node.find_child(NodeKind.LIST):
+        for list_item_node in list_node.find_child(NodeKind.LIST_ITEM):
+            example_data = defaultdict(str)
+
+            ref_nodes = find_and_remove_child(
+                list_item_node,
+                NodeKind.HTML,
+                lambda html_node: html_node.tag == "ref",
+            )
+            for ref_node in ref_nodes:
+                extract_reference(wxr, example_data, ref_node)
+
+            example_text = clean_node(wxr, {}, list_item_node.children)
+
+            senseid, example_text = match_senseid(example_text)
+
+            if example_text:
+                example_data["text"] = example_text
+
+            if senseid:
+                sense_data = [
+                    sense
+                    for sense in page_data[-1]["senses"]
+                    if sense["senseid"] == senseid
+                ]
+
+                for sense in sense_data:
+                    sense["examples"].append(example_data)
+
+            else:
+                if example_data:
+                    wxr.wtp.debug(
+                        f"Found example data without senseid and text: {example_data}",
+                        sortid="extractor/de/examples/extract_examples/28",
+                    )
+    for non_list_node in level_node.invert_find_child(NodeKind.LIST):
+        wxr.wtp.debug(
+            f"Found unexpected non-list node in example section: {non_list_node}",
+            sortid="extractor/de/examples/extract_examples/33",
+        )
+
+
+def extract_reference(
+    wxr: WiktextractContext, example_data: Dict[str, str], ref_node: WikiNode
+):
+    reference_data = defaultdict()
+
+    reference_data["raw_ref"] = clean_node(wxr, {}, ref_node.children)
```

Comment:
This line here is responsible for 90% of my debug statements. Yet, the debugging is done in `core/1021`. The all have message similar to this:

```Kaugummi/de_noun/Beispiele: DEBUG: too many args (3) in argument reference: ('Sammelwerk', '', '') at ['Kaugummi', 'Literatur']```

As far as I can tell this has nothing to do with the code I wrote but is somehow caused by the template definitions themselves. References are pretty standardized in the German Wiktionary and they all at one point call a template called "[Vorlage:Literatur](https://de.wiktionary.org/w/index.php?title=Vorlage:Literatur&action=edit)" that has a variable called Sammelwerk. 

I don't quite understand what's going on here. Do you perhaps have any suggestions what I could investigate?

Otherwise, I tend to ignore this issue since everything seems to work apart from the debug statement.

Reply:
You can ignore this debug log. I guess it's just for checking the length of template arguments isn't longer than 2, and it should ignore the empty strings.

Reply:
Alright. A bit annoying but otherwise not problematic.